src/test/java/org/openrewrite/kotlin/tree/FieldAccessTest.java

```diff
@@ -249,4 +253,51 @@ inner class B {
           )
         );
     }
+
+    @Issue("https://github.com/openrewrite/rewrite-kotlin/issues/517")
+    @Test
+    void nestedFieldAccessType() {
```

Comment:
The tree tests are meant for idempotent prints, and type mapping tests exist in KotlinTypeMapping.
Please move this to KotlinTypeMapping.

Note:
This is a revision to the test that doesn't involve transformations.
Notice that A, B, and C are each Identifiers, where types are being fixed.
That allows visitIdentifier to check each case.
The test adds a package to ensure that imports, which are also field access selects, do not contain types (I didn't add those assertions, but the types look good!)

```java
        @Issue("https://github.com/openrewrite/rewrite-kotlin/issues/517")
        @Test
        void nestedFieldAccessType() {
            rewriteRun(
              kotlin(
                """
                  package foo.bar
                  
                  class A {
                      class B {
                          class A {
                              class C
                          }
                      }
                  }
                  """
              ),
              kotlin(
                """
                  import foo.bar.A
                  
                  val x = A.B.A.C()
                  """,
                spec -> spec.afterRecipe(cu -> {
                      AtomicBoolean a1 = new AtomicBoolean(false);
                      AtomicBoolean a2 = new AtomicBoolean(false);
                      AtomicBoolean b = new AtomicBoolean(false);
                      AtomicBoolean c = new AtomicBoolean(false);
                      new KotlinIsoVisitor<Integer>() {
                          @Override
                          public J.Identifier visitIdentifier(J.Identifier identifier, Integer n) {
                              switch (identifier.getSimpleName()) {
                                  case "A": {
                                      if (identifier.getType() != null) {
                                          if ("".equals(identifier.getType().toString())) {
                                              a1.set(true);
                                          } else if ("".equals(identifier.getType().toString())) {
                                              a2.set(true);
                                          }
                                      }
                                  }
                                  case "B": {
                                      assertThat(identifier.getType().toString()).isEqualTo("");
                                      b.set(true);
                                  }
                                  case "C": {
                                      assertThat(identifier.getType().toString()).isEqualTo("");
                                      c.set(true);
                                  }
                              }
                              return super.visitIdentifier(identifier, n);
                          }
                      }.visit(cu, 0);
                      assertThat(a1.get()).isTrue();
                      assertThat(a2.get()).isTrue();
                      assertThat(b.get()).isTrue();
                      assertThat(c.get()).isTrue();
                  }
                )
              )
            );
        }
```

Reply:
talked offline, will keep it as is.